crates/ruff_python_semantic/src/model.rs

```diff
@@ -257,6 +257,24 @@ impl<'a> SemanticModel<'a> {
             .is_some_and(|binding| binding.kind.is_builtin())
     }
 
+    /// Return `true` if `member` is a reference to `builtins.$target`,
+    /// i.e. either `object` (where `object` is not overridden in the global scope),
+    /// or `builtins.object` (where `builtins` is imported as a module at the top level)
+    pub fn match_builtin_expr(&self, expr: &Expr, symbol: &str) -> bool {
+        debug_assert!(!symbol.contains('.'));
+        if !self.seen_module(Modules::BUILTINS) {
+            return match expr {
```

Comment:
I would remove this `return`, it's roughly what tripped me up in the first read. E.g., I find this clearer:

```rust
/// Return `true` if `member` is a reference to `builtins.$target`,
/// i.e. either `object` (where `object` is not overridden in the global scope),
/// or `builtins.object` (where `builtins` is imported as a module at the top level)
pub fn match_builtin_expr(&self, expr: &Expr, symbol: &str) -> bool {
    debug_assert!(!symbol.contains('.'));
    if self.seen_module(Modules::BUILTINS) {
        self.resolve_qualified_name(expr)
            .is_some_and(|qualified_name| match qualified_name.segments() {
                ["builtins" | "", member] => *member == symbol,
                _ => false,
            })
    } else {
        match expr {
            Expr::Name(ast::ExprName { id, .. }) => id == symbol && self.is_builtin(symbol),
            _ => false,
        }
    }
}
```